src/opentelemetry.erl

```diff
@@ -200,17 +202,45 @@ convert_timestamp(Timestamp, Unit) ->
     Offset = erlang:time_offset(),
     erlang:convert_time_unit(Timestamp + Offset, native, Unit).
 
--spec links([{TraceId, SpanId, Attributes, TraceState}]) -> links() when
+-spec links([{TraceId, SpanId, Attributes, TraceState} | span_ctx() | {span_ctx(), Attributes}]) -> links() when
       TraceId :: trace_id(),
       SpanId :: span_id(),
       Attributes :: attributes(),
       TraceState :: tracestate().
-links(List) ->
-    [link(TraceId, SpanId, Attributes, TraceState) || {TraceId, SpanId, Attributes, TraceState} <- List,
-                                                      is_integer(TraceId),
-                                                      is_integer(SpanId),
-                                                      is_list(Attributes),
-                                                      is_list(TraceState)].
+links(List) when is_list(List) ->
+    lists:filtermap(fun({TraceId, SpanId, Attributes, TraceState}) when is_integer(TraceId) ,
+                                                                        is_integer(SpanId) ,
+                                                                        is_list(Attributes) ,
+                                                                        is_list(TraceState) ->
+                            link_or_false(TraceId, SpanId, Attributes, TraceState);
+                       ({#span_ctx{trace_id=TraceId,
+                                   span_id=SpanId,
+                                   tracestate=TraceState}, Attributes}) when is_integer(TraceId) ,
+                                                                             is_integer(SpanId) ,
+                                                                             is_list(Attributes) ,
+                                                                             is_list(TraceState) ->
+                            link_or_false(TraceId, SpanId, Attributes, TraceState);
+                       (#span_ctx{trace_id=TraceId,
+                                  span_id=SpanId,
+                                  tracestate=TraceState}) when is_integer(TraceId) ,
+                                                               is_integer(SpanId) ,
+                                                               is_list(TraceState) ->
+                            link_or_false(TraceId, SpanId, [], TraceState);
+                       (_) ->
+                            false
+              end, List);
+links(_) ->
+    [].
+
+-spec link(span_ctx() | undefined) -> link().
```

Comment:
Isn't this function going to throw if `undefined` is passed? I can't seem to find a declaration that covers the case of `link(undefined, [])`

Reply:
Yup, fixing. Thanks!